### ISSUE TITLE
LogoutAsync silently fails

### ISSUE THREAD

**Issue by dracan** [auth0/auth0-oidc-client-net#19]
When calling `LogoutAsync`, if the logout redirect url hasn't been set up correctly, this logout will fail to work - but there's no indication that it's failed. LogoutAsync doesn't throw an exception, and there's no return value.

**Comment by nh43de** [comment]
Same problem - if user starts logout process but clicks cancel on the iOS native prompt, there is no way to handle this. Not suitable for production.

**Comment by Stavvor** [comment]
Any updates on this one?

**Comment by Stavvor** [comment]
@dracan @nh43de If you still need this change you can build iOS nuget package form this branch [https://github.com/Stavvor/auth0-oidc-client-net/tree/iOSworkaround](url)
`LogoutAsync` method is returning `BrowserResultType` so you can react properlly for usercancel or httperror

**Comment by Jehoel** [comment]
@Stavvor can you fix the link in your post? The text is correct, but the `href=""` is incorrect.

BTW, [I saw that you cancelled your PR](https://github.com/auth0/auth0-oidc-client-net/pull/25) to introduce this to the repo, why?

**Comment by Stavvor** [comment]
@Jehoel Thanks i fixed my link to point to the correct adress, I didn't mean to cancel this PR. It was probably an accident

**Comment by jsauve** [comment]
Big thanks for this. I was previously unable to react to user cancellation at the prompt.

**Comment by tonyyyrocks39** [comment]
has anybody got a fix on this...

When calling LogoutAsync, if the logout redirect url hasn't been set up correctly, this logout will fail to work - but there's no indication that it's failed. LogoutAsync doesn't throw an exception, and there's no return value.

**Comment by jsauve** [comment]
@tonyyyrocks39, the fix is in the master branch, but not in the nuget package 2.3.1 yet. You can build the master branch yourself to get the fix. That's what I did.
Tip: I had to convert the Auth0.OidcClient.Core project to a .Net Standard 2.0 project instead of .Net 4.6.

My guess is that it'll be in the nuget soon, but that's up to the Auth0 devs.

**Comment by jsauve** [comment]
Basically, LoginAsync() will now return a Task of BroswerResult. You can check BroswerResult.ResultType for ResultType.Sucess or ResultType.UserCancel.

**Comment by tonyyyrocks39** [comment]
Question was about Logoutasync(),in platformwebview in the line  return tcs.Task;(last line),it just stuck there ,it never returns a browerresult.
I guess this is the reason browser in not returning to the app again

**Comment by jsauve** [comment]
Is your redirect url setup correctly? Use the same redirect url for logout as you use for login.

The ones from the callback url section here:
https://auth0.com/docs/quickstart/native/xamarin/01-login

Makes sure it's in your custom urls in info.plist and equivalent in iOS. (Follow the docs)

Main this is: use the same callback/redirect url for logout as login.

**Comment by jsauve** [comment]
But again, if you're using the latest nuget (2.3.1) LogoutAsync() just retuns a plain Task. You'll need latest from master. And redirect url needs to be set in the Auth0 portal for your app.

**Comment by tonyyyrocks39** [comment]
Yes,Thanks for quick replies,but i have properly configured redirect url,problem is after that line tcs.Task it should ideally return to  return browserResult.ResultType; in Logoutasync which is not happening,may be i need to look into master branch if incase something got changed in platformview class

**Comment by jsauve** [comment]
Definitely try it if you're currently using the nuget. I have a few outstanding PRs waiting for review too. Check em out.

**Comment by tonyyyrocks39** [comment]
I just tried master branch,there also it seems have problem,browser does not return back to app

**Comment by jsauve** [comment]
If you're able to login to an identity provider successfully, but the browser fails to close afterwards, then there is something misconfigured in regards to the logoit call back url. So, let's clarify some things about your current setup:

1. Are your login call back url and logout callback url identical? They should be. And your callbacks should not contain "https". They should contain your iOS bundle identifier or your Android package name.

2. Have you ensured that the scheme of your callback url is entered in your info.plist on iOS and in the manifest in Android? This is *imperative* to the operating system knowing that it should close the browser when it encounters your custom url scheme.

Sorry if my questions seem excessive. I'm just fairly certain that something in regards to the callback url is misconfigured.

**Comment by jsauve** [comment]
...also, have you made sure that your logout callback url is entered in the " Allowed Logout URLs" field of your app in the Auth0 web portal for your Auth0 application? That's also imperative. This means your callback url is entered in TWO fields in the Auth0 portal, not ONLY for the

**Comment by tonyyyrocks39** [comment]
@jsauve if you have some time,please look into the issue#45,I have put my entire codes there.

**Comment by tonyyyrocks39** [comment]
Somehow I made it to work..But still facing problem,Currently when I debug line by line,logout is happening successfully.But when I directly hit F5 logout not working

**Comment by damieng** [comment]
Is there anything left actionable on this issue?  It's quite hard to follow.  Suggest we continue discussion outstanding remaining issue on #45 ?
